src/components/dashboard/main-content/course-enrollments/data/hooks.js

```diff
@@ -220,6 +239,14 @@ export const useCourseUpgradeData = ({
       return {
         ...defaultReturn,
         subsidyForCourse: applicableSubsidy,
+        redeem: async (e) => {
+          e?.preventDefault();
+          await sendEnterpriseTrackEventWithDelay(
+            enterpriseCustomer.uuid,
+            'edx.ui.enterprise.learner_portal.course.upgrade_button.subscription_license.clicked',
+          );
+          global.location.assign(applicableSubsidy.redemptionUrl);
+        },
       };
```

Comment:
[curious] Was this return not supposed to set `hasUpgradeAndConfirm` to true?

Reply:
Correct, audit -> verified upgrade with a subscription license does not use a distinct "Upgrade" CTA or have a confirmation modal so `hasUpgradeAndConfirm: false` from the `defaultReturn` is returned here.

Reply:
[addtl context] In the subscription license upgrade case, both the course title hyperlink and the "Start course" / "Resume" button CTA links through the Data Sharing Consent flow , without an explicit upgrade confirmation.